pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+            nest.SetStatus(
+                self.poisson_th, {'rate': self.stim_dict['th_rate'],
+                                  'start': self.stim_dict['th_start'],
+                                  'stop': self.stop_th})
+            nest.Connect(self.poisson_th, self.thalamic_population)
+            self.nr_synapses_th = synapses_th_matrix()
+            if self.K_scl != 1:
+                self.thalamic_weight = self.thalamic_weight/(self.K_scl**0.5)
+                self.nr_synapses_th = (self.nr_synapses_th * self.K_scl)
+        else:
+            print 'thalamic input not provided'
+
+    # poisson generators are created in this function
+    def create_poisson(self):
+        if self.net_dict['poisson_input']:
+            rate_ext = self.net_dict['bg_rate'] * self.K_ext
+            self.poisson = []
+            for i, target_pop in enumerate(self.pops):
+                    poisson = nest.Create('poisson_generator')
+                    nest.SetStatus(poisson, {'rate': rate_ext[i]})
+                    self.poisson.append(poisson)
+            print 'Poisson background input created'
+
+    # dc generators are created in this functions
+    def create_dc_generator(self):
+        if self.stim_dict['dc_input']:
+            dc_amp_stim = self.net_dict['K_ext'] * self.stim_dict['dc_amp']
+            self.dc = []
+            print 'DC_amp_stim', dc_amp_stim
+            for i, target_pop in enumerate(self.pops):
+                dc = nest.Create('dc_generator',
+                                 params={'amplitude': dc_amp_stim[i],
+                                         'start': self.stim_dict['dc_start'],
+                                         'stop': self.stim_dict['dc_start'] +
+                                         self.stim_dict['dc_dur']})
+                self.dc.append(dc)
+            print 'DC generator created'
+
+    # connections between neuronal populations are created in this function
+    def create_connections(self):
+        mean_delays = self.net_dict['mean_delay_matrix']
+        std_delays = self.net_dict['std_delay_matrix']
+        norm_clp = 'normal_clipped'
+        for i, target_pop in enumerate(self.pops):
+            for j, source_pop in enumerate(self.pops):
+                synapse_nr = int(self.synapses_scaled[i][j])
+                if synapse_nr >= 0.:
+                    weight = self.weight_mat[i][j]
+                    w_sd = abs(weight * self.weight_mat_std[i][j])
+                    conn_dict_poisson = {'rule': 'all_to_all'}
+                    syn_dict_poisson = {'model': 'static_synapse',
+                                        'weight': self.w_ext,
+                                        'delay': {'distribution': norm_clp,
+                                                  'mu': 1.5,
+                                                  'sigma': 0.75,
+                                                  'low': self.sim_resolution}}
```

Comment:
`conn_dict_poisson` and `syn_dic_poisson` are not used in this function and can thus be removed.